pilot/cmd/pilot-discovery/main.go

```diff
@@ -120,6 +120,8 @@ func init() {
 	serverArgs.MCPCredentialOptions.AttachCobraFlags(discoveryCmd)
 	discoveryCmd.PersistentFlags().IntVar(&serverArgs.MCPMaxMessageSize, "mcpMaxMsgSize", bootstrap.DefaultMCPMaxMsgSize,
 		"Max message size received by MCP's grpc client")
+	discoveryCmd.PersistentFlags().BoolVar(&serverArgs.IncrementalEDS, "incrementalEDS", false,
```

Comment:
I wonder if we should make this an environment variable rather than a flag? A couple reasons:

1) backward compatibility between pilot binary and injection parameters (this has bitten us in the past with adding flags to the sidecar)

2) No deprecation policy required for environment variables ... we can simply begin ignoring the environment variable if it exists.

@costinm @andraxylia WDYT?